src/translate_scale.rs

```diff
@@ -0,0 +1,158 @@
+        TranslateScale::new(Vec2::ZERO, s)
+    }
+
+    /// Create a new transformation with translation only.
+    #[inline]
+    pub fn translate(t: Vec2) -> TranslateScale {
+        TranslateScale::new(t, 1.0)
+    }
+
+    /// Decompose transformation into translation and scale.
+    pub fn as_tuple(self) -> (Vec2, f64) {
+        (self.translation, self.scale)
+    }
+}
+
+impl Default for TranslateScale {
+    #[inline]
+    fn default() -> TranslateScale {
+        TranslateScale::scale(1.0)
+    }
+}
+
+impl From<TranslateScale> for Affine {
+    fn from(ts: TranslateScale) -> Affine {
+        let TranslateScale { translation, scale } = ts;
+        Affine::new([scale, 0.0, 0.0, scale, translation.x, translation.y])
+    }
+}
+
+impl Mul<Point> for TranslateScale {
+    type Output = Point;
+
+    #[inline]
+    fn mul(self, other: Point) -> Point {
+        (self.scale * other.to_vec2()).to_point() + self.translation
+    }
+}
+
+impl Mul for TranslateScale {
+    type Output = TranslateScale;
+
+    #[inline]
+    fn mul(self, other: TranslateScale) -> TranslateScale {
+        TranslateScale {
+            translation: self.translation + self.scale * other.translation,
+            scale: self.scale * other.scale,
+        }
+    }
+}
+
+impl MulAssign for TranslateScale {
+    #[inline]
+    fn mul_assign(&mut self, other: TranslateScale) {
+        *self = self.mul(other);
+    }
+}
+
+impl Mul<TranslateScale> for f64 {
+    type Output = TranslateScale;
+
+    #[inline]
+    fn mul(self, other: TranslateScale) -> TranslateScale {
+        TranslateScale {
+            translation: other.translation * self,
+            scale: other.scale * self,
+        }
+    }
+}
+
+impl Add<Vec2> for TranslateScale {
+    type Output = TranslateScale;
+
+    #[inline]
+    fn add(self, other: Vec2) -> TranslateScale {
+        TranslateScale {
+            translation: self.translation + other,
+            scale: self.scale,
+        }
+    }
+}
+
+impl Add<TranslateScale> for Vec2 {
+    type Output = TranslateScale;
+
+    #[inline]
+    fn add(self, other: TranslateScale) -> TranslateScale {
+        other + self
+    }
+}
+
+impl AddAssign<Vec2> for TranslateScale {
+    #[inline]
+    fn add_assign(&mut self, other: Vec2) {
+        *self = self.add(other);
+    }
+}
+
+impl Sub<Vec2> for TranslateScale {
+    type Output = TranslateScale;
+
+    #[inline]
+    fn sub(self, other: Vec2) -> TranslateScale {
+        TranslateScale {
+            translation: self.translation - other,
+            scale: self.scale,
+        }
+    }
+}
+
+impl SubAssign<Vec2> for TranslateScale {
+    #[inline]
+    fn sub_assign(&mut self, other: Vec2) {
+        *self = self.sub(other);
+    }
+}
+
+impl Mul<Circle> for TranslateScale {
+    type Output = Circle;
+
+    #[inline]
+    fn mul(self, other: Circle) -> Circle {
+        Circle::new(self * other.center, self.scale * other.radius)
+    }
+}
+
+impl Mul<Rect> for TranslateScale {
```

Comment:
In general, these operations are not commutative. If you have a matrix A and a point x, you write "Ax" but not "xA". Certainly multiplication of two `TransformScale` transforms is not commutative (as opposed to origin-preserving scale), so I'm inclined to stick to the math here unless there's a strong argument.

Agree about adding more `Mul` implementations, but let's see what is reasonably easy...

Reply:
My motivations are practical: without some good reason, I don't think you should be able to write `transform * rect` without also being able to write `rect * transform`, especially since the latter form more closely matches the user†'s understanding of the operation (I am taking a rect and applying a transform.)

† (at least *this* user)

Reply:
I'm going to push back on this in the interest of staying true to the mathematical underpinnings, and try to address it in the docstring.

Reply:
Some considerations:

1. this is not a matrix operation
2. The output of the multiply is explicit; there can be no ambiguity about whether `vec2 * transform`  is returning a new vector or a new transform.
3. There should be(?) no ambiguity about the possible output of these operations, if implemented.
4. Math is not a good place to look for *ergonomics* inspiration.
5. As a user, I don't want to go read the docs to figure out why `a * t` works but `t * a` doesn't, when they appear, (to me) to be trivially equivalent.

Anyway, docs are better than nothing, but I'm still not convinced leaving these out is better.

To think about this another way: what is a possible error, confusion, or bug that could arise from the inclusion of these impls?

Reply:
I think (1) is where we disagree. To me, this is a matrix operation, just with additional constraints.

I was going to write something more philosophical about a pedagogical role of kurbo, and an aesthetic goal of capturing the mathematical ideals, but I think it's more useful to answer that last question directly. By conforming to the rules for matrix math, it lets you confidently apply the standard mathematical transformations such as multiplication being associative.

Not allowing point * transform seems basically the same as not allowing scalar * point or point + point, and I think a similar case could be made that these are useful (I know I write `to_vec` and `to_point` a lot to get around this).